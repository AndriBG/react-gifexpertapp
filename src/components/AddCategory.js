import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [ inputValue, setInputValue ] = useState('');

    const handleInputValue = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>  {
        e.preventDefault();

        if ( inputValue.trim().length > 1 ) {
            setCategories(catgs => [inputValue, ...catgs]);
            setInputValue('');
        }

    }

    return (
        <form onSubmit= { handleSubmit }>
            {/* <p>{ inputValue }</p> */}
            <input
                type = 'Text'
                placeholder = 'Buscar'
                value = { inputValue }
                onChange = { handleInputValue }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
