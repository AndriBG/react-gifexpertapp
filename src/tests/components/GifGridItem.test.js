import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en el componente GifGridItem', () => {

    const title = 'Un t√≠tulo';
    const url = 'Https://localhost/algo.png';

    const wrapper = shallow(
        <GifGridItem 
            title= {title}
            url= {url}
        />
    );

    test('<GifGridItem /> debe estar correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe tener el p igual que el title', () => {

        const p = wrapper.find('p');

        expect( p.text() ).toBe( title );

    });

    test('debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.props().alt ).toBe( title );

    });

})


