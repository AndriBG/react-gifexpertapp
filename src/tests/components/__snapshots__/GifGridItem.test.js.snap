// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente GifGridItem <GifGridItem /> debe estar correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un título"
    src="Https://localhost/algo.png"
  />
  <p>
    Un título
  </p>
</div>
`;
